modules/nextflow/src/test/groovy/nextflow/config/ConfigBuilderTest.groovy

```diff
@@ -475,12 +479,13 @@ class ConfigBuilderTest extends Specification {
         when:
         def opt = new CliOptions()
         def run = new CmdRun(paramsFile: params, params: [alpha: 'Hola', beta: 'Mundo'])
-        def result = new ConfigBuilder().setOptions(opt).setCmdRun(run).buildGivenFiles(file)
+        def baseDir = Files.createTempDirectory('assets')
+        def result = new ConfigBuilder().setOptions(opt).setCmdRun(run).setBaseDir(baseDir).buildGivenFiles(file)
 
         then:
         result.params.alpha == 'Hola'   // <-- this comes from the CLI
         result.params.beta == 'Mundo'   // <-- this comes from the CLI as well
-        result.params.omega == 'Last'   // <-- this comes from the params-file
+        result.params.omega == "${baseDir.toUriString()}/data/omega.txt"   // <-- this comes from the params-file and replaces parameters
         result.params.gamma == "I'm gamma"   // <-- from the config
```

Comment:
Here the `baseDir` should not be resolved to the actual value?

Reply:
Yes, it is resolved to the actual value. Notice that I'm using double quotes. At this test `baseDir` is a temporary directory that I cannot hardcode beforehand, this is why I'm using the variable itself.